packages/flutter/lib/src/material/dialog.dart

```diff
@@ -510,11 +522,25 @@ class AlertDialog extends StatelessWidget {
     const double m3ActionEndPadding = 18.0;
     const double m3ActionBottomPadding = 12.0;
 
+    Widget? iconWidget;
     Widget? titleWidget;
     Widget? contentWidget;
     Widget? actionsWidget;
+
+    if (icon != null) {
+      iconWidget = Padding(
+        padding: const EdgeInsets.fromLTRB(24.0, 24.0, 24.0, 0.0),
```

Comment:
nit: 24 top, 16 bottom here, and 0 top for `defaultTitlePadding` below when there's an icon

Reply:
makes sense, done